### ISSUE TITLE
FEATURE: Add sourcegraph source

### ISSUE THREAD

**Issue by ghostbuster91** [nvim-neo-tree/neo-tree.nvim#1244]
### Did you check the docs?

- [X] I have read all the docs.

### Is your feature request related to a problem? Please describe.

Not really a problem. I like to show tries in a consistent way no matter where they come from. 
I've just came across sourcegraph [plugin](https://github.com/sourcegraph/sg.nvim) for neovim and it seems to have a builtin file tree explorer. 

### Describe the solution you'd like.

Implement new source for sg.nvim. It should mimic the behavior of the sg's builtin file explorer. 

### Describe alternatives you've considered.

This can be a separate project, but I wanted to start somewhere and this repository seemed like a good choice.

**Comment by ghostbuster91** [comment]
Yes, I agree with you that it would be the best to make it a separate project. Just wanted to share this idea. Maybe a discussion would have been a better place. Feel free to move it there.

**Comment by cseickel** [comment]
Yes, an external source would be the best way to go about this. Here is the example repo: https://github.com/nvim-neo-tree/example-source

It is a very simple hello world. Netman or the internal filesystem source would be good real world examples of interactive lazy loaded file system browsing.

I'm going to convert this to a discussion.
